src/core/derivation.ts

```diff
@@ -119,23 +119,24 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T) {
 	derivation.runId = ++globalState.runId;
 	const prevTracking = globalState.trackingDerivation;
 	globalState.trackingDerivation = derivation;
-	let hasException = true;
+	let hasException = false;
 	let result: T;
 	try {
 		result = f.call(derivation);
-		hasException = false;
+	} catch(e){
+		hasException = true;
+		handleExceptionInDerivation(derivation, e);
+		throw e;
```

Comment:
The stack trace is generated when `new Error` is called, so rethrowing the same error object shouldn't change the stack (unless `handleExceptionInDerivation` mutates the error object)